oak-run/src/main/java/org/apache/jackrabbit/oak/indexversion/IndexVersionOperation.java

```diff
@@ -86,42 +92,65 @@ public static List<IndexVersionOperation> generateIndexVersionOperationList(Node
         }
 
         if (!reverseSortedIndexNameList.isEmpty()) {
-            IndexName activeIndexNameObject = reverseSortedIndexNameList.remove(0);
-            NodeState activeIndexNode = indexDefParentNode.getChildNode(PathUtils.getName(activeIndexNameObject.getNodeName()));
-            boolean isActiveIndexLongEnough = isIndexPurgeReady(activeIndexNameObject, activeIndexNode, purgeThresholdMillis);
-            int activeProductVersion = activeIndexNameObject.getProductVersion();
-            indexVersionOperationList.add(new IndexVersionOperation(activeIndexNameObject));
+            IndexName activeIndexNameObject = getActiveIndex(reverseSortedIndexNameList, parentPath, rootNode);
+            if (activeIndexNameObject == null) {
+                LOG.warn("Cannot find any active index from the list: {}", reverseSortedIndexNameList);
```

Comment:
I think it's fine to log a warning currently... I just wonder, how would we remove an OOTB index from AEM. The first such case is the generic Lucene index (/oak:index/lucene). We can remove all versions of that index in GraniteContent. So there would never be this case that he have indexes, but none is active (the old state is: active; the new state is: removed).

Reply:
@thomasmueller good points, yeah makes sense just to leave a warning message as is for now. we can check further later based on the log analysis.
For now, it will be too risky to simply remove all indexes when no active index.